### PR TITLE
feat: implement ErrorHandler & LoopWrapper & RowButtons, improve KeyboardSetYAML, fix: ServerConnectionError, docs: Option monad

### LINKED ISSUE #40: fix ServerDisconnectedError while bot is running
Sometimes a `aiohttp.client_exceptions.ServerDisconnectedError` exception occurs while the bot is running.

Error log

### PR DESCRIPTION
**Pull request by luwqz1**
closes #40

### CODE REVIEW

**Comment by timoniq** [comment]
Is it finished now?

**Comment by luwqz1** [comment]
> Is it finished now?

yeah. you can do a merge

File: telegrinder/option/msgspec_option.py [hunk 1]
```diff
@@ -22,39 +22,40 @@ def __init__(self, value: Value | None = None) -> None:
...
-        return self.__get_variant().map_or(default, f)
+        return self.as_option_variant.map_or(default, f)
 
     def map_or_else(self, default: typing.Callable[[], T], f: typing.Callable[[Value], T], /) -> T:
-        return self.__get_variant().map_or_else(default, f)
+        return self.as_option_variant.map_or_else(default, f)
 
     def expect(self, error: str | BaseException, /) -> Value:
-        return self.__get_variant().expect(error)
+        return self.as_option_variant.expect(error)
```
Comment (timoniq) [inline telegrinder/option/msgspec_option.py:58]: honestly i dont like neither the name for this thing nor the intrication itself. is there anything we can do about it?

File: docs/tools/option.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# Option usage instead of typing.Optional:
+class Admin(Model):
+    name: str
+    age: int
+    bio: Option[str] = Option.Nothing
+
+
+async def main():
+    me = await api.get_me()
+    print("my last name is", me.last_name.unwrap_or("Unknown"))
```
Comment (timoniq) [inline docs/tools/option.md:25]: looks cool! ö

**Comment by timoniq** [comment]
Thanks for PR! I added some minor comments, as we resolve them I'll be able to merge up these thingies

**Comment by luwqz1** [comment]
> Thanks for PR! I added some minor comments, as we resolve them I'll be able to merge up these thingies

It was nothing ^_^
I've resolved the problems you mentioned in the comments!

**Review by timoniq — APPROVED** [review]

**Pull request merged**